src/gui/plugins/plotting/Plotting.cc

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <ignition/plugin/Register.hh>
+
+#include "Plotting.hh"
+
+namespace ignition::gazebo
+{
+  class PlottingPrivate
+  {
+    /// \brief Interface to communicate with Qml
+    public: ignition::gui::PlottingInterface *plottingIface;
+
+    public: std::map<std::string, PlotComponent*> components;
+  };
+
+  class PlotComponentPrivate
+  {
+    /// \brief entity id in the simulation
+    public: uint64_t entity;
```

Comment:
I think you can use `Entity` instead of `uint64_t`. This way, if we ever change `Entity`, this will be updated at the same time. You don't need to change the signals and slots.